### PR TITLE
Update stream endpoint to support a userId param

### LINKED ISSUE #256: Scope `/api/stream` to a specific user
Return all streams by a user. Should do the same thing that `/api/stream/user/<userId>` currently does.

The `200` response for the new endpoint and `/api/stream/user/<userId>` should be the same

This may be relevant https://github.com/livepeer/livepeer-com/blob/master/packages/api/src/schema/schema.yaml

### LINKED ISSUE #257: Update Livepeer.com dashboard stream list to use `/api/stream?userId=*`
https://livepeer.com/app/user

### LINKED ISSUE #258: Confirm Livepeer.com admin streams list still works
/app/admin/streams

### PR DESCRIPTION
**Pull request by julianbenegas**
**What does this pull request do? Explain your changes. (required)**

- `/api/stream` now supports `userId` query param, that filters streams by that userId.
- User streams table now hits `/api/stream`. The old endpoint remains unchanged for backwards compatibility.

**Does this pull request close any open issues?**

- closes #256 
- closes #257
- closes #258

**Screenshots (optional):**

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/stream.js [hunk 1]
```diff
@@ -101,11 +101,19 @@ app.get("/", authMiddleware({ admin: true }), async (req, res) => {
     nonLivepeerOnly,
     order,
     filters,
+    userId,
   } = req.query;
   if (isNaN(parseInt(limit))) {
     limit = undefined;
   }
 
+  if (userId && req.user.admin !== true && req.user.id !== userId) {
```
Comment (darkdarkdragon) [inline packages/api/src/controllers/stream.js]: This whole endpoint is accessible only for admins (`authMiddleware({ admin: true })`)
So you should remove `{admin: true}` to make it accessible to regular users.

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

File: packages/api/src/controllers/stream.js (hunk 1 above)
Comment (wohlner) [reply packages/api/src/controllers/stream.js]: Great! Glad we can easily open up this endpoint to regular users. I'm sure you are already doing this, but I want to make sure that a regular users only has access to their own streams, not everyones. Only admins should see all users' streams.

Comment (julianbenegas) [reply packages/api/src/controllers/stream.js]: Yup, that is handled.

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]
[unit test](https://github.com/livepeer/livepeer-com/blob/f2a421f962144babc46e8e944b6a3b50377439c6/packages/api/src/controllers/stream.test.js#L380) failing

**Comment by julianbenegas** [comment]
> [unit test](https://github.com/livepeer/livepeer-com/blob/f2a421f962144babc46e8e944b6a3b50377439c6/packages/api/src/controllers/stream.test.js#L380) failing

thanks @darkdarkdragon, will fix

File: packages/api/src/controllers/stream.test.js [hunk 2]
```diff
@@ -373,8 +373,8 @@ describe("controllers/stream", () => {
       expect(res.status).toBe(403);
     });
 
-    it("should not get all streams for admin user", async () => {
-      client.apiKey = adminApiKey;
+    it("should not get all streams for non admin user", async () => {
```
Comment (julianbenegas) [inline packages/api/src/controllers/stream.test.js:376]: @darkdarkdragon changed the test to test for `non admin user`.

It's a bit weird, the `authMiddleware({ admin: true })` creates the need for a JWT, but the endpoint as is doesn't require it. Do you think I should update the permissions for it to require a JWT? 🤔

Comment (darkdarkdragon) [reply packages/api/src/controllers/stream.test.js:376]: > Do you think I should update the permissions for it to require a JWT?

No, there is no need.

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]
`all` and `nonLivepeerOnly` arguments should be ignored for non-admin users

**Comment by wohlner** [comment]
@julianbenegas This is now deployed on stage to livepeer.monster. I just tested a `GET` request to https://livepeer.monster/api/stream using a token for a non-admin .monster account and got this error `{"errors":["user can only request information on their own streams"]}`. 

And then when I made the same GET request using an admin .monster token, the request response is super slow/doesn't work. I got a 524 error, the response timed out. 

Both these `GET` requests to `/api/stream` should return a list of all the API key holder's `stream` objects.

cc:@darkdragon#2215

**Pull request merged**